src/charm.py

```diff
@@ -1234,6 +1353,25 @@ def _run_diagnostic_command(self, cmd) -> None:
         except subprocess.CalledProcessError as e:
             logger.error(f"Exception occurred running '{cmd}'\n {e}")
 
+    def _reconfigure_replica_set(self) -> None:
+        """Optionally reconfigure replicaset if needed.
+
+        This should happen only when we are reusing storage from one cluster to another cluster,
+        because all IPs have changed and all configurations are broken.
```

Comment:
please include in the docstring the exceptions this raises

QQ: does this raise NotReadyException?

Reply:
It does not raise NotReadyException:
 * We don't want to check for sync in progress here as we'll only update in the case none of the IPs in the config are part of the IPs in the databag.